packages/core/src/stateUtils.ts

```diff
@@ -1616,11 +1631,12 @@ export function macrostep(
         status: 'stopped'
       }
     );
-    states.push(nextSnapshot);
+    // states.push(nextSnapshot);
+    addMicrostate(nextSnapshot, event, []);
```

Comment:
I wonder if this wouldn't be better served by some "stop" event - this call site is unique to state machines but what about stopped promises etc?

Reply:
What might this look like?

Reply:
`@xstate.actor` is already produced for all actors in `class Actor`. `'xstate.stop'` always originates in that same `class Actor` (in its `_stop`). So maybe this could just be moved closer to `Actor`? That would give us some symmetry around this

Reply:
I'm not sure how to accomplish this, but feel free to give it a try - I'm open to it